### PR TITLE
Switch to ESLint

### PR DESCRIPTION
**Pull request by wwlorey**

### CODE REVIEW

File: test/parseDetectorTables.test.ts [hunk 1]
```diff
@@ -8,18 +8,21 @@ import { ColumnName, detectorTable, findTableByName, getValuesByColumnName, vali
 import { createTestContext, ITestContext } from './global.test';
 
 suite('Detector Dataset Parser', () => {
-    test('Find table by table name', async () => {
+    /* eslint-disable @typescript-eslint/no-unsafe-member-access */
```
Comment (ejizba) [inline test/parseDetectorTables.test.ts:11]: Can you do a test-specific eslint file just like we had a test-specifc tslint.json? I would disable these "any" rules for the whole test folder, just like it used to be.

File: package.json [hunk 2]
```diff
@@ -812,8 +812,8 @@
         "cleanReadme": "gulp cleanReadme",
         "compile": "tsc -watch",
         "package": "vsce package --githubBranch main",
-        "lint": "eslint -c .eslintrc.js --ext .ts .",
-        "lint-fix": "eslint -c .eslintrc.js --ext .ts . --fix",
+        "lint": "eslint --ext .ts .",
+        "lint-fix": "eslint --ext .ts . --fix",
```
Comment (wwlorey) [inline package.json:816]: For ESLint to use the `.eslintrc.js` in `test/` I had to remove the config file option here

File: test/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+module.exports = {
```
Comment (ejizba) [inline test/.eslintrc.js:1]: I'm pretty sure there's a way you can remove the duplicate code and get this based off the root `.eslintrc.js`

This doc talks about using "extends" and also about the fact that these are literally javascript files, so you might be able to work some magic related to that: https://eslint.org/docs/developer-guide/shareable-configs

**Review by ejizba — APPROVED** [review]
🎉

**Pull request merged**